### ISSUE TITLE
Webjob Not Renewing Certificates

### ISSUE THREAD

**Issue by vinsuz** [sjkp/letsencrypt-siteextension#100]
Hi Sjkp,
First of all, thank you. Between Let's Encrypt and your Azure site extension, our company has saved a tremendous amount of resources.

Everything is working great for us except the renewal process. At first, the webjobs weren't running, so when I receive an expiry notification that's the first thing I check. But even for the apps where the webjob is running, it is failing to renew the certificate. Here's the log for the last attempt:

`[02/03/2017 07:54:54 > 51bc4a: INFO] Executing: 'Functions.RenewCertificate' - Reason: 'Timer fired at 2017-02-03T07:54:53.7420229+00:00'`
`[02/03/2017 07:54:55 > 51bc4a: INFO] Renew certificate`
`[02/03/2017 07:54:59 > 51bc4a: INFO] Completed renewal of '0' certificates`
`[02/03/2017 07:54:59 > 51bc4a: INFO] Executed: 'Functions.RenewCertificate' (Succeeded)`
`[02/03/2017 08:54:29 > 51bc4a: SYS INFO] WebJob is still running`

Notice the "Completed renewal of '0' certificates"? The certificate is due to expire on February 7th, well within the timeframe for renewal. **This particular domain is running the Azure Let's Encrypt (x86) Version 0.5.0 extension. Is this an issue of simply upgrading to Version 5.5?**

Thanks again, and let me know if there's somewhere I can contribute.

Vin

**Comment by sjkp** [comment]
I just did a quick investigation, it doesn't fail for me, and I don't remember there being any updates to that part since of the codebase since 0.5.0. So upgrading shouldn't fix the problem if there is one. 

I will have to provide a build with more logging information, or your maybe if you can, first upgrade and then run the job while have application logging turned on, and provide me with the output from that run, in case it still fails.
![image](https://cloud.githubusercontent.com/assets/3397744/22621751/1c363f24-eb2b-11e6-8207-f37ec15b4c2e.png)

The application logs end up in D:\home\LogFiles\Application which you can access using the kudu portal.

**Comment by vinsuz** [comment]
Just saw this interesting tid bit:

> 'Always On' doesn't appear to be enabled for this Web App. To ensure your continuous job doesn't stop running when the SCM host is idle for too long, consider enabling 'Always On' in the configuration settings for your Web App.

Interesting, eh?

**Comment by sjkp** [comment]
Yes if you are not running always on and there is no traffic to your site, then the webjobs will not run. Unless you are on the free plan (where you can't activate always on) I would recommend that you enable it, it is a bad user experience for those users that visit your site, after it has been spun down as it will be very slow for them.

**Comment by robrich** [comment]
I just hit this too.  Please consider updating the install wiki with this step.

**Comment by ohadschn** [comment]
I think it's also possible to use Azure Scheduler:
https://spr.com/creating-a-scheduled-background-webjob-in-microsoft-azure/

Azure Scheduler no longer has a free tier, but it still comes out much cheaper than the _Basic_ App Service tier (which I believe is the minimum for _Always On_).
